### ISSUE TITLE
`toString()` completion not suggested for types

### ISSUE THREAD

**Issue by gayalkuruppu** [ballerina-platform/ballerina-lang#36052]
**Description:**
$Title

**Steps to reproduce:**
```
function foo() {
    string bar = int.<cur>
}
```
We don't get any completions here though following is a valid usage for a type

```
function foo() {
    string bar = int.toBalString();
    string car = int.toString();
}
```

**Affected Versions:**
2201.1.1

**Comment by IMS94** [comment]
@gayalkuruppu You have to use `int:<cursor>` to get `int` langlib's completions. Also, the `toString()` method comes from the `value` langlib. So it should be `value:toString()`.

```ballerina
int a = 10;
string str = a.toString();
```
In the above example, `a.toString()` is possible because an int inherits the langlib methods from `value`.

**Comment by gayalkuruppu** [comment]
@IMS94  Agree. My original concern was to explore the usages of the completion `Thread` at the cursor in the following scenario. 
```
string a = <cur>
```
As discussed offline, `string a = Thread.toString()` is one such usage. Hence the issue.

**Comment by IMS94** [comment]
@gayalkuruppu  regarding the following case: 
> ```
> function foo() {
>     string bar = int.<cur>
> }
> ```
> 
> We don't get any completions here though following is a valid usage for a type
> 
> ```
> function foo() {
>     string bar = int.toBalString();
>     string car = int.toString();
> }
> ```

`int.<cursor>` is not a valid usage. Valid usage is `int:<cursor>`.

**Comment by IMS94** [comment]
Just checked this, The compiler allows `string a = int.toString();`. First we have to verify what `toString()` refers to here. Is it the `toString()` method of the type itself?  If so, we should provide such completions.

Regarding `Thread` being suggested. `Thread` is available in visible symbols. And it's sorted to the top because `Thread` is a union of 2 literals. This we should fix when checking assignability.

**Comment by IMS94** [comment]
Sorting of `Thread` type has been addressed with #35935

**Comment by IMS94** [comment]
`int.toString()`  outputs `typedesc int` and has no use case. Since `int:toString()` or `<inv_var>.toString()` is the accepted usage, LS has no plan to provide completions for the `int` type. Therefore, closing this issue.

**Comment by IMS94** [comment]
```ballerina
typedesc foo = int;
    string _ = foo.<cursor>
```
We provide completions here. Which means, we should do the same for `int.<cursor>`. Reopening.

**Comment by IMS94** [comment]
I implemented a possible solution by updating the `FieldAccessCompletionResolver` to override the `transform(BuiltinSimpleNameReferenceNode node)` method and return `TypeDescriptorTypeSymbol` of `int` from that in this case.

Next when the langlibs were taken from that type symbol. While trying to generate the labels for completion items, I encountered an `Illegal State Exception: symbol is 'null'` exception from the `typeIds(any|error)` langlib method's `signature()` method. Inside that, it tries to get the `definition()` of the `error` (param) type and there we have a place that access the `getModule()` of the parameter type symbol: See `BallerinaErrorTypeSymbol.java:138`.

It eventually calls `io.ballerina.compiler.api.impl.SymbolFactory#getBCompiledSymbol()` method that throws the illegal state exception. @sanjana can you have a look?
